pulser-core/pulser/json/abstract_repr/deserializer.py

```diff
@@ -263,30 +264,46 @@ def _deserialize_channel(obj: dict[str, Any]) -> Channel:
         params["eom_config"] = None
         if obj["eom_config"] is not None:
             data = obj["eom_config"]
-            params["eom_config"] = RydbergEOM(
-                mod_bandwidth=data["mod_bandwidth"],
-                limiting_beam=RydbergBeam[data["limiting_beam"]],
-                max_limiting_amp=data["max_limiting_amp"],
-                intermediate_detuning=data["intermediate_detuning"],
-                controlled_beams=tuple(
-                    RydbergBeam[beam] for beam in data["controlled_beams"]
-                ),
-            )
+            try:
+                params["eom_config"] = RydbergEOM(
+                    mod_bandwidth=data["mod_bandwidth"],
+                    limiting_beam=RydbergBeam[data["limiting_beam"]],
+                    max_limiting_amp=data["max_limiting_amp"],
+                    intermediate_detuning=data["intermediate_detuning"],
+                    controlled_beams=tuple(
+                        RydbergBeam[beam] for beam in data["controlled_beams"]
+                    ),
+                )
+            except ValueError as e:
+                raise AbstractReprError(
+                    "RydbergEOM deserialization failed."
+                ) from e
     elif obj["basis"] == "digital":
         channel_cls = Raman
     elif obj["basis"] == "XY":
         channel_cls = Microwave
+    # No other basis allowed by the schema
 
     for param in dataclasses.fields(channel_cls):
         if param.init and param.name != "eom_config":
             params[param.name] = obj[param.name]
-    return channel_cls(**params)
+    try:
+        ch_obj = channel_cls(**params)
```

Comment:
nit: no need to create a ch_obj variable you can directly return it here, and the exception will be caught regardless